go-controller/pkg/ovn/gress_policy.go

```diff
@@ -171,7 +182,23 @@ func (gp *gressPolicy) deletePeerPod(pod *v1.Pod) error {
 		klog.Infof("Failed to get pod IPs %s/%s to delete from network policy peers: %w", pod.Namespace, pod.Name, err)
 		return nil
 	}
-	return gp.peerAddressSet.DeleteIPs(ips)
+
```

Comment:
It needs additional synchronization, because add and delete may be called for different pods (with the same ip) at the same time. So we need to update `gp.ipToPodOwnership` and the real address set atomically
You can try local syncmap package to lock per-ip

Reply:
oh even unit tests catched this, nice!

Reply:
Changed from `map` to `syncmap.SyncMap`. @npinaeva Please, take a look.